client/src/components/SettingsDialog.tsx

```diff
@@ -0,0 +1,86 @@
+import { Container, SegmentedControl, Select } from "@mantine/core";
+import { useClickOutside } from "@mantine/hooks";
+import React from "react";
+import useHTMLElementList from "../hooks/useHTMLElementList";
+import {
+  ThemeType,
+  usePreferencesActions,
+  useTheme,
+  useUnitPreferences,
+} from "../stores/PreferencesStore";
+import { stylesWithThemedBackgroundColor } from "../utils/styles";
+import {
+  MeasurementTypes,
+  DistanceUnits,
+  SpeedUnits,
+  AngleUnits,
+} from "../utils/units";
+
+interface SettingsDialogProps {
+  onClose?: () => void;
+}
+
+export default function SettingsDialog({ onClose }: SettingsDialogProps) {
+  const theme = useTheme();
+  const unitPreferences = useUnitPreferences();
+
+  const { updateUnitPreferences, updateTheme } = usePreferencesActions();
+
+  const { register, data: internalHTMLNodes } = useHTMLElementList();
+
+  useClickOutside(() => onClose?.(), undefined, internalHTMLNodes);
+
+  return (
+    <Container
+      mt="xs"
+      p="xs"
+      sx={(theme) => stylesWithThemedBackgroundColor(theme)}
+      {...register("settingsContainer")}
```

Comment:
I got a little cute here. It was 1:30am when I wrote it so it may be weird, but I was trying to do something similar to React Hook Form's registration API